board/safety/safety_ford.h

```diff
@@ -82,16 +82,16 @@ static int ford_tx_hook(CANPacket_t *to_send, bool longitudinal_allowed) {
     tx = 0;
   }
 
-  // Cruise button check, only allow cancel button to be sent
-  if (addr == MSG_STEERING_DATA_FD1) {
-    // Violation if any button other than cancel is pressed
-    // Signal: CcAslButtnCnclPress
-    bool violation = (GET_BYTE(to_send, 0) |
-                      (GET_BYTE(to_send, 1) & 0xFEU) |
-                      GET_BYTE(to_send, 2) |
-                      GET_BYTE(to_send, 3) |
-                      GET_BYTE(to_send, 4) |
-                      GET_BYTE(to_send, 5)) != 0U;
+  // Cruise button check to only allow cancel button press when controls are off
+  if ((addr == MSG_STEERING_DATA_FD1) && !controls_allowed) {
+    // Violation if any set/resume button is pressed while controls not allowed
+    //   CcAslButtnCnclPress is 0x01 in byte 1 (not violation)
+    //   CcAsllButtnResPress is 0x02 in byte 3 (violation)
+    //   TjaButtnOnOffPress is 0x01 in byte 5 (not violation)
+    bool violation = ((GET_BYTE(to_send, 1) & 0x06U) |
+                      (GET_BYTE(to_send, 2) & 0x21U) |
+                      (GET_BYTE(to_send, 3) & 0xFEU) |
+                      (GET_BYTE(to_send, 4) & 0xE0U)) != 0U;
```

Comment:
If they're 1 bit signals, you can use GET_BIT, with the msb from the DBC

Reply:
decided to just check the signals that we send in OP instead